### PR TITLE
Add optional custom pinnumbers to connector

### PR DESCRIPTION
**Pull request by kimmoli**
One possible solution for custom pin numbering, e.g. fits needs in #4 

- Pincount is detected automatically if not given.
- If both, pinout and pinnumbers, are given, their count must match.
- Freedom to give pin numbers in any order, in alphabetical and mixed notation.

Sample

```
connectors:
  X1:
    type: D-Sub
    subtype: male
    pincount: 25
    pinnumbers: [1,14,3,16,5,18,7,20,9,22,11,24,13]
    pinout: [ SENSE_P_1, SENSE_N_1, SENSE_P_2, SENSE_N_2, SENSE_P_3, SENSE_N_3, SENSE_P_4,SENSE_N_4, SENSE_P_5, SENSE_N_5, SENSE_P_6, SENSE_N_6, GND ]
  X2:
    type: F48
    subtype: female
    pincount: 48
    pinnumbers: [ z2,b2,d2,z4,b4,d4,z6,b6,d6,z8,b8,d8,z10,b10,d10,z12,b12,d12,z14,b14,d14,z16,b16,d16,z18,b18,d18,z20,b20,d20,z22,b22,d22,z24,b24,d24,z26,b26,d26,z28,b28,d28,z30,b30,d30,z32,b32,d32 ]

cables:
  W1:
    gauge: 0.25 mm2
    length: 0.2
    color_code: DIN
    wirecount: 16
    shield: true

connections:
  -
    - X1: [1,14,3,16,5,18,7,20,9,22,11,24]
    - W1: [2,1,4,3,6,5,8,7,10,9,12,11]
    - X2: [d4,z2,d10,z8,d16,z14,d20,z18,d26,z24,d32,z30]
  -
    - X1: 13
    - W1: s

```
demo01

### CODE REVIEW

File: src/wireviz.py [hunk 1]
```diff
@@ -327,15 +335,17 @@ def __post_init__(self):
...
+            if self.pinout:
                 self.pincount = len(self.pinout)
-        else:
-            if not self.pincount:
-                self.pincount = 1
-            self.pinout = ['',] * self.pincount
+            elif self.pinnumbers:
+                self.pincount = len(self.pinnumbers)
+            else:
+                raise Exception('You need to specify at least one, pincount, pinout or pincount')
```
Comment (elliotmr) [inline src/wireviz.py:344]: I guess one of these should be pinnumbers

Comment (kimmoli) [reply src/wireviz.py]: sure. fixed.

**Comment by formatc1702** [comment]
This is a fantastic addition, I'll take care of it in the next few days!

**Comment by formatc1702** [comment]
Unfortunately, this breaks the minimalist way of specifying only a pincount (e.g. Tutorial 01)

**Comment by formatc1702** [comment]
Idea: Perhaps, instead of having `pinout` and `pinnumbers`, using `pin_labels` and `pin_descriptions` or something similar might be less confusing/ambiguous? Not sure if `pinnumbers` works when it's not only numerical labels...

**Comment by kimmoli** [comment]
Thanks for your efforts on this project.

> 
> 
> Additional idea: Perhaps, instead of having `pinout` vs `pinnumbers`, using `pin_labels` or `pin_IDs` vs `pin_descriptions`, `pin_functions` or similar might be less confusing/ambiguous? Not sure if `pinnumbers` works when it's not only numerical labels...

IMHO `pinnumber` is pin number, regardles is it number, letter or mixed.

pinout could be `pinname`

i would avoid underscores ...

**Comment by formatc1702** [comment]
`pinnumber` and `pinname` could work. I totally agree on the underscore thing. Thanks for helping out!

**Comment by kvid** [comment]
IMHO `pinlabel` is more intuitive when the values might be non-numeric, and it works very well wirhout underscore.

**Comment by kvid** [comment]
IMHO `pinnumbers` might be confusing when the values might be non-numeric. Why not use `pins` as suggested in issue#4 if these new pin identifiers will be the only way to reference pins? I would  also consider the possability to reference pins in alternative ways, like `#3` for the third pin and `$GND` for the pin named GND in the pinout, etc. The `pinlabels` and `pinnames` alternatives can also be considered without the underscore.

**Comment by kvid** [comment]
I have not tested the proposed code to see if it is possible to specify `s` as pin identifier and how then a conflict with the shield identifier is handeled. Is this an issue?

**Pull request merged**